![Valentine Days](imagem\valentine.png)

About Dataset

Valentine's Day embodies the spirit of love and affection. Valentine's Day heralds the midpoint of February when couples will exchange flowers and chocolates, head out for a romantic candlelit dinner and celebrate their enduring love. Nevertheless not everyone purchase these valentine gifts.
This dataset can be used to identify the purchasing habit of different people age, salary, gender, and location of residence
Inspiration: Valentine Mood-
From the valentine mood, I just decided to have data for analysis

- Customer ID: This is customers unique identifier
- Gender: Customer's gender, female or male
- Age: Customer's age
- Location: Where The customer is based
- EstimatedIncome: Customer estimated income
- Purchased: Gift purchased or not

URL Dataset: https://www.kaggle.com/datasets/kanyianalyst/valentine-gift-purchase


In [1]:
# Versão da Linguagem Python
from platform import python_version
print('Versão da Linguagem Python Usada Neste Jupyter Notebook:', python_version())

Versão da Linguagem Python Usada Neste Jupyter Notebook: 3.9.13


In [2]:
# Instala o pacote watermark. 
# Esse pacote é usado para gravar as versões de outros pacotes usados neste jupyter notebook.
!pip install -q -U watermark

In [3]:
# Instala o pacote iPython-sql 
!pip install -q ipython-sql

In [4]:
# Imports
import pandas as pd
import sqlite3

In [5]:
# Versões dos pacotes usados neste jupyter notebook
%reload_ext watermark
%watermark -a "Data Science Academy" --iversions

Author: Data Science Academy

pandas : 1.5.3
sqlite3: 2.6.0



In [6]:
# Criamos a conexão a um banco de dados SQLite
cnn = sqlite3.connect('database/Valentine_Gift_Purchase.db')

In [7]:
# Cria um dataframe
dfvalentine = pd.read_csv('database/Valentine_Gift_Purchase.csv')

In [8]:
# Copia o dataframe para dentro do banco de dados como uma tabela
dfvalentine.to_sql('valentine_gift_purchase', cnn)

420

In [9]:
# Carregamos a extensão SQL
%load_ext sql

In [10]:
# Definimos o banco de dados
%sql sqlite:///database/Valentine_Gift_Purchase.db

# <p align="center"><strong>Limpeza e processamento de dados</strong></p>

##### Abaixo estão as técnicas utilizadas para limpar e processar dados das variáveis Customer ID, Gender, Age, Location, Estimated Income e Purchased.

- Tratamentos de valores ausentes
- Detecção e remoção de registros duplicados
- Reorganização dos dados
- Transformação de atributos e parsing
- Filtragem e agregação
- Imputação




In [11]:
%%sql

-- Este é um comentário de linha em SQL no Jupyter Notebook

SELECT * FROM valentine_gift_purchase

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(0, 'CID00211', 'Male', 19.0, 'Rural', 22000.0, 0),
 (1, 'CID00212', 'Male', 35.0, 'Semi Arban', 20000.0, 0),
 (2, 'CID00213', 'Female', 26.0, 'Arban', 36000.0, 0),
 (3, 'CID00214', 'Female', 27.0, 'Rural', 57000.0, 0),
 (4, 'CID00215', 'Male', 19.0, 'Rural', 76000.0, 0),
 (5, 'CID00216', 'Male', 27.0, 'Rural', 60000.0, 1),
 (6, 'CID00217', 'Female', 27.0, 'Rural', 84000.0, 0),
 (7, 'CID00218', 'Female', 32.0, 'Arban', 150000.0, 1),
 (8, 'CID00219', 'Male', 25.0, 'Rural', 33000.0, 1),
 (9, 'CID00220', 'Female', 35.0, 'Rural', 65000.0, 1),
 (10, 'CID00221', 'Female', 26.0, 'Rural', 80000.0, 0),
 (11, 'CID00222', 'Female', 26.0, 'Rural', 52000.0, 0),
 (12, 'CID00223', 'Male', None, 'Rural', 86000.0, 1),
 (13, 'CID00224', 'Male', 32.0, 'Semi Arban', 19000.0, 1),
 (14, 'CID00225', 'Male', 18.0, 'Semi Arban', 82000.0, 1),
 (15, 'CID00226', 'Male', 29.0, 'Semi Arban', 80000.0, 1),
 (16, 'CID00227', 'Male', 47.0, 'Arban', None, 1),
 (17, 'CID00227', 'Male', 47.0, 'Arban', 25000.0, 1),
 (18, 'CID00228', 'Male', 45.0, 'Arban', 26000.0, 1),
 (19, 'CID00228', 'Male', 45.0, 'Arban', 26000.0, 1),
 (20, 'CID00229', 'Male', 46.0, 'Arban', 28000.0, 1),
 (21, 'CID00229', 'Male', 46.0, 'Arban', 28000.0, 1),
 (22, 'CID00230', 'Female', 48.0, 'Arban', 29000.0, 0),
 (23, 'CID00231', 'Male', 45.0, 'Semi Arban', 22000.0, 1),
 (24, 'CID00232', 'Female', 47.0, 'Semi Arban', 49000.0, 0),
 (25, 'CID00233', 'Male', None, 'Semi Arban', 41000.0, 0),
 (26, 'CID00234', 'Female', 45.0, 'Semi Arban', 22000.0, 0),
 (27, 'CID00235', 'Male', 46.0, 'Arban', 23000.0, 1),
 (28, 'CID00236', 'Male', 47.0, 'Arban', 20000.0, 1),
 (29, 'CID00237', 'Male', 49.0, 'Semi Arban', 28000.0, 1),
 (30, 'CID00238', 'Female', 47.0, 'Rural', 30000.0, 1),
 (31, 'CID00239', 'Male', 29.0, 'Rural', 43000.0, 1),
 (32, 'CID00240', 'Male', 31.0, 'Rural', 18000.0, 0),
 (33, 'CID00241', 'Male', 31.0, 'Rural', 74000.0, 1),
 (34, 'CID00242', 'Female', 27.0, 'Rural', 137000.0, 1),
 (35, 'CID00243', 'Female', 21.0, 'Rural', 16000.0, 0),
 (36, 'CID00244', 'Female', 28.0, 'Rural', 44000.0, 0),
 (37, 'CID00245', 'Male', 27.0, 'Rural', 90000.0, 1),
 (38, 'CID00246', 'Male', 35.0, 'Rural', 27000.0, 0),
 (39, 'CID00247', 'Female', 33.0, 'Rural', 28000.0, 0),
 (40, 'CID00248', 'Male', None, 'Rural', 49000.0, 0),
 (41, 'CID00249', 'Female', 26.0, 'Rural', 72000.0, 0),
 (42, 'CID00250', 'Female', 27.0, 'Rural', 31000.0, 0),
 (43, 'CID00251', 'Female', 27.0, 'Rural', 17000.0, 0),
 (44, 'CID00252', 'Female', 33.0, 'Rural', 51000.0, 0),
 (45, 'CID00253', 'Male', 35.0, 'Rural', 108000.0, 1),
 (46, 'CID00254', 'Male', 30.0, 'Semi Arban', 15000.0, 0),
 (47, 'CID00255', 'Female', 28.0, 'Semi Arban', 84000.0, 0),
 (48, 'CID00256', 'Male', 23.0, 'Semi Arban', 20000.0, 0),
 (49, 'CID00257', 'Male', 25.0, 'Arban', 79000.0, 1),
 (50, 'CID00258', 'Female', 27.0, 'Arban', 54000.0, 0),
 (51, 'CID00259', 'Male', 30.0, 'Arban', 135000.0, 1),
 (52, 'CID00260', 'Female', 31.0, 'Arban', 89000.0, 1),
 (53, 'CID00261', 'Female', 24.0, 'Arban', 32000.0, 1),
 (54, 'CID00262', 'Female', 18.0, 'Arban', 44000.0, 1),
 (55, 'CID00263', 'Female', None, 'Arban', 83000.0, 0),
 (56, 'CID00264', 'Female', 35.0, 'Arban', 23000.0, 1),
 (57, 'CID00265', 'Female', 27.0, 'Arban', 58000.0, 1),
 (58, 'CID00266', 'Female', 24.0, 'Arban', 55000.0, 1),
 (59, 'CID00267', 'Female', 23.0, 'Arban', 48000.0, 1),
 (60, 'CID00268', 'Male', 28.0, 'Rural', 79000.0, 1),
 (61, 'CID00269', 'Male', 22.0, 'Rural', 18000.0, 0),
 (62, 'CID00270', 'Female', 32.0, 'Rural', 117000.0, 1),
 (63, 'CID00271', 'Male', 27.0, 'Rural', 20000.0, 0),
 (64, 'CID00272', 'Male', 25.0, 'Rural', 87000.0, 1),
 (65, 'CID00273', 'Female', 23.0, 'Rural', 66000.0, 0),
 (66, 'CID00274', 'Male', 32.0, 'Semi Arban', 120000.0, 1),
 (67, 'CID00275', 'Female', None, 'Rural', 83000.0, 0),
 (68, 'CID00276', 'Male', 24.0, 'Rural', 58000.0, 1),
 (69, 'CID00277', 'Male', 24.0, 'Rural', 19000.0, 0),
 (70, 'CID00278', 'Female', 23.0, 'Rural', 82000.0, 1),
 (71, 'CID00279', 'Female', 22.0, 'Rural',

In [12]:
%%sql

-- Atualiza os dados retirand todas as iniciais CID do CUSTOMER ID 

UPDATE valentine_gift_purchase SET "Customer ID" = REPLACE("Customer ID", 'CID','')
WHERE "Customer ID" LIKE 'CID%';


 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[]

In [13]:
%%sql

-- Retorna todos os registros armazenados nessa tabela

SELECT * FROM valentine_gift_purchase

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(0, '00211', 'Male', 19.0, 'Rural', 22000.0, 0),
 (1, '00212', 'Male', 35.0, 'Semi Arban', 20000.0, 0),
 (2, '00213', 'Female', 26.0, 'Arban', 36000.0, 0),
 (3, '00214', 'Female', 27.0, 'Rural', 57000.0, 0),
 (4, '00215', 'Male', 19.0, 'Rural', 76000.0, 0),
 (5, '00216', 'Male', 27.0, 'Rural', 60000.0, 1),
 (6, '00217', 'Female', 27.0, 'Rural', 84000.0, 0),
 (7, '00218', 'Female', 32.0, 'Arban', 150000.0, 1),
 (8, '00219', 'Male', 25.0, 'Rural', 33000.0, 1),
 (9, '00220', 'Female', 35.0, 'Rural', 65000.0, 1),
 (10, '00221', 'Female', 26.0, 'Rural', 80000.0, 0),
 (11, '00222', 'Female', 26.0, 'Rural', 52000.0, 0),
 (12, '00223', 'Male', None, 'Rural', 86000.0, 1),
 (13, '00224', 'Male', 32.0, 'Semi Arban', 19000.0, 1),
 (14, '00225', 'Male', 18.0, 'Semi Arban', 82000.0, 1),
 (15, '00226', 'Male', 29.0, 'Semi Arban', 80000.0, 1),
 (16, '00227', 'Male', 47.0, 'Arban', None, 1),
 (17, '00227', 'Male', 47.0, 'Arban', 25000.0, 1),
 (18, '00228', 'Male', 45.0, 'Arban', 26000.0, 1),
 (19, '00228', 'Male', 45.0, 'Arban', 26000.0, 1),
 (20, '00229', 'Male', 46.0, 'Arban', 28000.0, 1),
 (21, '00229', 'Male', 46.0, 'Arban', 28000.0, 1),
 (22, '00230', 'Female', 48.0, 'Arban', 29000.0, 0),
 (23, '00231', 'Male', 45.0, 'Semi Arban', 22000.0, 1),
 (24, '00232', 'Female', 47.0, 'Semi Arban', 49000.0, 0),
 (25, '00233', 'Male', None, 'Semi Arban', 41000.0, 0),
 (26, '00234', 'Female', 45.0, 'Semi Arban', 22000.0, 0),
 (27, '00235', 'Male', 46.0, 'Arban', 23000.0, 1),
 (28, '00236', 'Male', 47.0, 'Arban', 20000.0, 1),
 (29, '00237', 'Male', 49.0, 'Semi Arban', 28000.0, 1),
 (30, '00238', 'Female', 47.0, 'Rural', 30000.0, 1),
 (31, '00239', 'Male', 29.0, 'Rural', 43000.0, 1),
 (32, '00240', 'Male', 31.0, 'Rural', 18000.0, 0),
 (33, '00241', 'Male', 31.0, 'Rural', 74000.0, 1),
 (34, '00242', 'Female', 27.0, 'Rural', 137000.0, 1),
 (35, '00243', 'Female', 21.0, 'Rural', 16000.0, 0),
 (36, '00244', 'Female', 28.0, 'Rural', 44000.0, 0),
 (37, '00245', 'Male', 27.0, 'Rural', 90000.0, 1),
 (38, '00246', 'Male', 35.0, 'Rural', 27000.0, 0),
 (39, '00247', 'Female', 33.0, 'Rural', 28000.0, 0),
 (40, '00248', 'Male', None, 'Rural', 49000.0, 0),
 (41, '00249', 'Female', 26.0, 'Rural', 72000.0, 0),
 (42, '00250', 'Female', 27.0, 'Rural', 31000.0, 0),
 (43, '00251', 'Female', 27.0, 'Rural', 17000.0, 0),
 (44, '00252', 'Female', 33.0, 'Rural', 51000.0, 0),
 (45, '00253', 'Male', 35.0, 'Rural', 108000.0, 1),
 (46, '00254', 'Male', 30.0, 'Semi Arban', 15000.0, 0),
 (47, '00255', 'Female', 28.0, 'Semi Arban', 84000.0, 0),
 (48, '00256', 'Male', 23.0, 'Semi Arban', 20000.0, 0),
 (49, '00257', 'Male', 25.0, 'Arban', 79000.0, 1),
 (50, '00258', 'Female', 27.0, 'Arban', 54000.0, 0),
 (51, '00259', 'Male', 30.0, 'Arban', 135000.0, 1),
 (52, '00260', 'Female', 31.0, 'Arban', 89000.0, 1),
 (53, '00261', 'Female', 24.0, 'Arban', 32000.0, 1),
 (54, '00262', 'Female', 18.0, 'Arban', 44000.0, 1),
 (55, '00263', 'Female', None, 'Arban', 83000.0, 0),
 (56, '00264', 'Female', 35.0, 'Arban', 23000.0, 1),
 (57, '00265', 'Female', 27.0, 'Arban', 58000.0, 1),
 (58, '00266', 'Female', 24.0, 'Arban', 55000.0, 1),
 (59, '00267', 'Female', 23.0, 'Arban', 48000.0, 1),
 (60, '00268', 'Male', 28.0, 'Rural', 79000.0, 1),
 (61, '00269', 'Male', 22.0, 'Rural', 18000.0, 0),
 (62, '00270', 'Female', 32.0, 'Rural', 117000.0, 1),
 (63, '00271', 'Male', 27.0, 'Rural', 20000.0, 0),
 (64, '00272', 'Male', 25.0, 'Rural', 87000.0, 1),
 (65, '00273', 'Female', 23.0, 'Rural', 66000.0, 0),
 (66, '00274', 'Male', 32.0, 'Semi Arban', 120000.0, 1),
 (67, '00275', 'Female', None, 'Rural', 83000.0, 0),
 (68, '00276', 'Male', 24.0, 'Rural', 58000.0, 1),
 (69, '00277', 'Male', 24.0, 'Rural', 19000.0, 0),
 (70, '00278', 'Female', 23.0, 'Rural', 82000.0, 1),
 (71, '00279', 'Female', 22.0, 'Rural', 63000.0, 1),
 (72, '00280', 'Female', 31.0, 'Semi Arban', 68000.0, 0),
 (73, '00281', 'Male', 25.0, 'Semi Arban', 80000.0, 0),
 (74, '00282', 'Female', 24.0, 'Semi Arban', 27000.0, 0),
 (75, '00283', 'Female', 20.0,

In [14]:
%%sql

-- Verifica se tem valores nulo

SELECT COUNT(*) 
FROM valentine_gift_purchase
WHERE "Customer ID" is null 

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(0,)]

In [15]:
%%sql

-- Verifica total de valores

SELECT  "Customer ID", COUNT(*) AS total
FROM valentine_gift_purchase
GROUP BY "Customer ID"
ORDER BY total DESC;


 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[('00229', 2),
 ('00228', 2),
 ('00227', 2),
 ('00627', 1),
 ('00626', 1),
 ('00625', 1),
 ('00624', 1),
 ('00623', 1),
 ('00622', 1),
 ('00621', 1),
 ('00620', 1),
 ('00619', 1),
 ('00618', 1),
 ('00617', 1),
 ('00616', 1),
 ('00615', 1),
 ('00614', 1),
 ('00613', 1),
 ('00612', 1),
 ('00611', 1),
 ('00610', 1),
 ('00609', 1),
 ('00608', 1),
 ('00607', 1),
 ('00606', 1),
 ('00605', 1),
 ('00604', 1),
 ('00603', 1),
 ('00602', 1),
 ('00601', 1),
 ('00600', 1),
 ('00599', 1),
 ('00598', 1),
 ('00597', 1),
 ('00596', 1),
 ('00595', 1),
 ('00594', 1),
 ('00593', 1),
 ('00592', 1),
 ('00591', 1),
 ('00590', 1),
 ('00589', 1),
 ('00588', 1),
 ('00587', 1),
 ('00586', 1),
 ('00585', 1),
 ('00584', 1),
 ('00583', 1),
 ('00582', 1),
 ('00581', 1),
 ('00580', 1),
 ('00579', 1),
 ('00578', 1),
 ('00577', 1),
 ('00576', 1),
 ('00575', 1),
 ('00574', 1),
 ('00573', 1),
 ('00572', 1),
 ('00571', 1),
 ('00570', 1),
 ('00569', 1),
 ('00568', 1),
 ('00567', 1),
 ('00566', 1),
 ('00565', 1),
 ('00564', 1),
 ('00563', 1),
 ('00562', 1),
 ('00561', 1),
 ('00560', 1),
 ('00559', 1),
 ('00558', 1),
 ('00557', 1),
 ('00556', 1),
 ('00555', 1),
 ('00554', 1),
 ('00553', 1),
 ('00552', 1),
 ('00551', 1),
 ('00550', 1),
 ('00549', 1),
 ('00548', 1),
 ('00547', 1),
 ('00546', 1),
 ('00545', 1),
 ('00544', 1),
 ('00543', 1),
 ('00542', 1),
 ('00541', 1),
 ('00540', 1),
 ('00539', 1),
 ('00538', 1),
 ('00537', 1),
 ('00536', 1),
 ('00535', 1),
 ('00534', 1),
 ('00533', 1),
 ('00532', 1),
 ('00531', 1),
 ('00530', 1),
 ('00529', 1),
 ('00528', 1),
 ('00527', 1),
 ('00526', 1),
 ('00525', 1),
 ('00524', 1),
 ('00523', 1),
 ('00522', 1),
 ('00521', 1),
 ('00520', 1),
 ('00519', 1),
 ('00518', 1),
 ('00517', 1),
 ('00516', 1),
 ('00515', 1),
 ('00514', 1),
 ('00513', 1),
 ('00512', 1),
 ('00511', 1),
 ('00510', 1),
 ('00509', 1),
 ('00508', 1),
 ('00507', 1),
 ('00506', 1),
 ('00505', 1),
 ('00504', 1),
 ('00503', 1),
 ('00502', 1),
 ('00501', 1),
 ('00500', 1),
 ('00499', 1),
 ('00498', 1),
 ('00497', 1),
 ('00496', 1),
 ('00495', 1),
 ('00494', 1),
 ('00493', 1),
 ('00492', 1),
 ('00491', 1),
 ('00490', 1),
 ('00489', 1),
 ('00488', 1),
 ('00487', 1),
 ('00486', 1),
 ('00485', 1),
 ('00484', 1),
 ('00483', 1),
 ('00482', 1),
 ('00481', 1),
 ('00480', 1),
 ('00479', 1),
 ('00478', 1),
 ('00477', 1),
 ('00476', 1),
 ('00475', 1),
 ('00474', 1),
 ('00473', 1),
 ('00472', 1),
 ('00471', 1),
 ('00470', 1),
 ('00469', 1),
 ('00468', 1),
 ('00467', 1),
 ('00466', 1),
 ('00465', 1),
 ('00464', 1),
 ('00463', 1),
 ('00462', 1),
 ('00461', 1),
 ('00460', 1),
 ('00459', 1),
 ('00458', 1),
 ('00457', 1),
 ('00456', 1),
 ('00455', 1),
 ('00454', 1),
 ('00453', 1),
 ('00452', 1),
 ('00451', 1),
 ('00450', 1),
 ('00449', 1),
 ('00448', 1),
 ('00447', 1),
 ('00446', 1),
 ('00445', 1),
 ('00444', 1),
 ('00443', 1),
 ('00442', 1),
 ('00441', 1),
 ('00440', 1),
 ('00439', 1),
 ('00438', 1),
 ('00437', 1),
 ('00436', 1),
 ('00435', 1),
 ('00434', 1),
 ('00433', 1),
 ('00432', 1),
 ('00431', 1),
 ('00430', 1),
 ('00429', 1),
 ('00428', 1),
 ('00427', 1),
 ('00426', 1),
 ('00425', 1),
 ('00424', 1),
 ('00423', 1),
 ('00422', 1),
 ('00421', 1),
 ('00420', 1),
 ('00419', 1),
 ('00418', 1),
 ('00417', 1),
 ('00416', 1),
 ('00415', 1),
 ('00414', 1),
 ('00413', 1),
 ('00412', 1),
 ('00411', 1),
 ('00410', 1),
 ('00409', 1),
 ('00408', 1),
 ('00407', 1),
 ('00406', 1),
 ('00405', 1),
 ('00404', 1),
 ('00403', 1),
 ('00402', 1),
 ('00401', 1),
 ('00400', 1),
 ('00399', 1),
 ('00398', 1),
 ('00397', 1),
 ('00396', 1),
 ('00395', 1),
 ('00394', 1),
 ('00393', 1),
 ('00392', 1),
 ('00391', 1),
 ('00390', 1),
 ('00389', 1),
 ('00388', 1),
 ('00387', 1),
 ('00386', 1),
 ('00385', 1),
 ('00384', 1),
 ('00383', 1),
 ('00382', 1),
 ('00381', 1),
 ('00380', 1),
 ('00379', 1),
 ('00378', 1),
 ('00377', 1),
 ('00376', 1),
 ('00375', 1),
 ('00374', 1),
 ('00373', 1),
 ('00372', 1),
 ('00371', 1),
 ('00370', 1),
 ('00369', 1),
 ('00368', 1),
 ('00367', 1),
 ('00366', 1),
 ('00365', 1),
 ('00364',

In [16]:
%%sql 

-- Identificando valores duplicados

SELECT "Customer ID", "Gender", Age, "Location", EstimatedIncome, Purchased, count(*) as duplicados
FROM valentine_gift_purchase
group by "Customer ID", "Gender", Age, "Location", EstimatedIncome, Purchased
having duplicados >1;

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[('00228', 'Male', 45.0, 'Arban', 26000.0, 1, 2),
 ('00229', 'Male', 46.0, 'Arban', 28000.0, 1, 2)]

In [17]:
%%sql 

-- Deletando valores duplicados

DELETE FROM valentine_gift_purchase
WHERE ROWID NOT IN (
    SELECT MIN(ROWID)
    FROM valentine_gift_purchase
    WHERE "Customer ID" IS NOT NULL AND EstimatedIncome IS NOT NULL
    GROUP BY "Customer ID"
)
AND EstimatedIncome IS NULL;


 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[]

In [18]:
%%sql

-- Identificando valores duplicados

SELECT  "Customer ID", COUNT(*) AS total
FROM valentine_gift_purchase
GROUP BY "Customer ID"
ORDER BY total DESC;


 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[('00229', 2),
 ('00228', 2),
 ('00627', 1),
 ('00626', 1),
 ('00625', 1),
 ('00624', 1),
 ('00623', 1),
 ('00622', 1),
 ('00621', 1),
 ('00620', 1),
 ('00619', 1),
 ('00618', 1),
 ('00617', 1),
 ('00616', 1),
 ('00615', 1),
 ('00614', 1),
 ('00613', 1),
 ('00612', 1),
 ('00611', 1),
 ('00610', 1),
 ('00609', 1),
 ('00608', 1),
 ('00607', 1),
 ('00606', 1),
 ('00605', 1),
 ('00604', 1),
 ('00603', 1),
 ('00602', 1),
 ('00601', 1),
 ('00600', 1),
 ('00599', 1),
 ('00598', 1),
 ('00597', 1),
 ('00596', 1),
 ('00595', 1),
 ('00594', 1),
 ('00593', 1),
 ('00592', 1),
 ('00591', 1),
 ('00590', 1),
 ('00589', 1),
 ('00588', 1),
 ('00587', 1),
 ('00586', 1),
 ('00585', 1),
 ('00584', 1),
 ('00583', 1),
 ('00582', 1),
 ('00581', 1),
 ('00580', 1),
 ('00579', 1),
 ('00578', 1),
 ('00577', 1),
 ('00576', 1),
 ('00575', 1),
 ('00574', 1),
 ('00573', 1),
 ('00572', 1),
 ('00571', 1),
 ('00570', 1),
 ('00569', 1),
 ('00568', 1),
 ('00567', 1),
 ('00566', 1),
 ('00565', 1),
 ('00564', 1),
 ('00563', 1),
 ('00562', 1),
 ('00561', 1),
 ('00560', 1),
 ('00559', 1),
 ('00558', 1),
 ('00557', 1),
 ('00556', 1),
 ('00555', 1),
 ('00554', 1),
 ('00553', 1),
 ('00552', 1),
 ('00551', 1),
 ('00550', 1),
 ('00549', 1),
 ('00548', 1),
 ('00547', 1),
 ('00546', 1),
 ('00545', 1),
 ('00544', 1),
 ('00543', 1),
 ('00542', 1),
 ('00541', 1),
 ('00540', 1),
 ('00539', 1),
 ('00538', 1),
 ('00537', 1),
 ('00536', 1),
 ('00535', 1),
 ('00534', 1),
 ('00533', 1),
 ('00532', 1),
 ('00531', 1),
 ('00530', 1),
 ('00529', 1),
 ('00528', 1),
 ('00527', 1),
 ('00526', 1),
 ('00525', 1),
 ('00524', 1),
 ('00523', 1),
 ('00522', 1),
 ('00521', 1),
 ('00520', 1),
 ('00519', 1),
 ('00518', 1),
 ('00517', 1),
 ('00516', 1),
 ('00515', 1),
 ('00514', 1),
 ('00513', 1),
 ('00512', 1),
 ('00511', 1),
 ('00510', 1),
 ('00509', 1),
 ('00508', 1),
 ('00507', 1),
 ('00506', 1),
 ('00505', 1),
 ('00504', 1),
 ('00503', 1),
 ('00502', 1),
 ('00501', 1),
 ('00500', 1),
 ('00499', 1),
 ('00498', 1),
 ('00497', 1),
 ('00496', 1),
 ('00495', 1),
 ('00494', 1),
 ('00493', 1),
 ('00492', 1),
 ('00491', 1),
 ('00490', 1),
 ('00489', 1),
 ('00488', 1),
 ('00487', 1),
 ('00486', 1),
 ('00485', 1),
 ('00484', 1),
 ('00483', 1),
 ('00482', 1),
 ('00481', 1),
 ('00480', 1),
 ('00479', 1),
 ('00478', 1),
 ('00477', 1),
 ('00476', 1),
 ('00475', 1),
 ('00474', 1),
 ('00473', 1),
 ('00472', 1),
 ('00471', 1),
 ('00470', 1),
 ('00469', 1),
 ('00468', 1),
 ('00467', 1),
 ('00466', 1),
 ('00465', 1),
 ('00464', 1),
 ('00463', 1),
 ('00462', 1),
 ('00461', 1),
 ('00460', 1),
 ('00459', 1),
 ('00458', 1),
 ('00457', 1),
 ('00456', 1),
 ('00455', 1),
 ('00454', 1),
 ('00453', 1),
 ('00452', 1),
 ('00451', 1),
 ('00450', 1),
 ('00449', 1),
 ('00448', 1),
 ('00447', 1),
 ('00446', 1),
 ('00445', 1),
 ('00444', 1),
 ('00443', 1),
 ('00442', 1),
 ('00441', 1),
 ('00440', 1),
 ('00439', 1),
 ('00438', 1),
 ('00437', 1),
 ('00436', 1),
 ('00435', 1),
 ('00434', 1),
 ('00433', 1),
 ('00432', 1),
 ('00431', 1),
 ('00430', 1),
 ('00429', 1),
 ('00428', 1),
 ('00427', 1),
 ('00426', 1),
 ('00425', 1),
 ('00424', 1),
 ('00423', 1),
 ('00422', 1),
 ('00421', 1),
 ('00420', 1),
 ('00419', 1),
 ('00418', 1),
 ('00417', 1),
 ('00416', 1),
 ('00415', 1),
 ('00414', 1),
 ('00413', 1),
 ('00412', 1),
 ('00411', 1),
 ('00410', 1),
 ('00409', 1),
 ('00408', 1),
 ('00407', 1),
 ('00406', 1),
 ('00405', 1),
 ('00404', 1),
 ('00403', 1),
 ('00402', 1),
 ('00401', 1),
 ('00400', 1),
 ('00399', 1),
 ('00398', 1),
 ('00397', 1),
 ('00396', 1),
 ('00395', 1),
 ('00394', 1),
 ('00393', 1),
 ('00392', 1),
 ('00391', 1),
 ('00390', 1),
 ('00389', 1),
 ('00388', 1),
 ('00387', 1),
 ('00386', 1),
 ('00385', 1),
 ('00384', 1),
 ('00383', 1),
 ('00382', 1),
 ('00381', 1),
 ('00380', 1),
 ('00379', 1),
 ('00378', 1),
 ('00377', 1),
 ('00376', 1),
 ('00375', 1),
 ('00374', 1),
 ('00373', 1),
 ('00372', 1),
 ('00371', 1),
 ('00370', 1),
 ('00369', 1),
 ('00368', 1),
 ('00367', 1),
 ('00366', 1),
 ('00365', 1),
 ('00364', 1),
 ('00363',

In [19]:
%%sql 

-- Deletando valores duplicados

DELETE FROM valentine_gift_purchase
WHERE ROWID NOT IN (
    SELECT MIN(ROWID)
    FROM valentine_gift_purchase
    GROUP BY "Customer ID"
);


 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[]

In [20]:
%%sql
-- Retorna todos os registros armazenados nessa tabela

SELECT * FROM valentine_gift_purchase

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(0, '00211', 'Male', 19.0, 'Rural', 22000.0, 0),
 (1, '00212', 'Male', 35.0, 'Semi Arban', 20000.0, 0),
 (2, '00213', 'Female', 26.0, 'Arban', 36000.0, 0),
 (3, '00214', 'Female', 27.0, 'Rural', 57000.0, 0),
 (4, '00215', 'Male', 19.0, 'Rural', 76000.0, 0),
 (5, '00216', 'Male', 27.0, 'Rural', 60000.0, 1),
 (6, '00217', 'Female', 27.0, 'Rural', 84000.0, 0),
 (7, '00218', 'Female', 32.0, 'Arban', 150000.0, 1),
 (8, '00219', 'Male', 25.0, 'Rural', 33000.0, 1),
 (9, '00220', 'Female', 35.0, 'Rural', 65000.0, 1),
 (10, '00221', 'Female', 26.0, 'Rural', 80000.0, 0),
 (11, '00222', 'Female', 26.0, 'Rural', 52000.0, 0),
 (12, '00223', 'Male', None, 'Rural', 86000.0, 1),
 (13, '00224', 'Male', 32.0, 'Semi Arban', 19000.0, 1),
 (14, '00225', 'Male', 18.0, 'Semi Arban', 82000.0, 1),
 (15, '00226', 'Male', 29.0, 'Semi Arban', 80000.0, 1),
 (17, '00227', 'Male', 47.0, 'Arban', 25000.0, 1),
 (18, '00228', 'Male', 45.0, 'Arban', 26000.0, 1),
 (20, '00229', 'Male', 46.0, 'Arban', 28000.0, 1),
 (22, '00230', 'Female', 48.0, 'Arban', 29000.0, 0),
 (23, '00231', 'Male', 45.0, 'Semi Arban', 22000.0, 1),
 (24, '00232', 'Female', 47.0, 'Semi Arban', 49000.0, 0),
 (25, '00233', 'Male', None, 'Semi Arban', 41000.0, 0),
 (26, '00234', 'Female', 45.0, 'Semi Arban', 22000.0, 0),
 (27, '00235', 'Male', 46.0, 'Arban', 23000.0, 1),
 (28, '00236', 'Male', 47.0, 'Arban', 20000.0, 1),
 (29, '00237', 'Male', 49.0, 'Semi Arban', 28000.0, 1),
 (30, '00238', 'Female', 47.0, 'Rural', 30000.0, 1),
 (31, '00239', 'Male', 29.0, 'Rural', 43000.0, 1),
 (32, '00240', 'Male', 31.0, 'Rural', 18000.0, 0),
 (33, '00241', 'Male', 31.0, 'Rural', 74000.0, 1),
 (34, '00242', 'Female', 27.0, 'Rural', 137000.0, 1),
 (35, '00243', 'Female', 21.0, 'Rural', 16000.0, 0),
 (36, '00244', 'Female', 28.0, 'Rural', 44000.0, 0),
 (37, '00245', 'Male', 27.0, 'Rural', 90000.0, 1),
 (38, '00246', 'Male', 35.0, 'Rural', 27000.0, 0),
 (39, '00247', 'Female', 33.0, 'Rural', 28000.0, 0),
 (40, '00248', 'Male', None, 'Rural', 49000.0, 0),
 (41, '00249', 'Female', 26.0, 'Rural', 72000.0, 0),
 (42, '00250', 'Female', 27.0, 'Rural', 31000.0, 0),
 (43, '00251', 'Female', 27.0, 'Rural', 17000.0, 0),
 (44, '00252', 'Female', 33.0, 'Rural', 51000.0, 0),
 (45, '00253', 'Male', 35.0, 'Rural', 108000.0, 1),
 (46, '00254', 'Male', 30.0, 'Semi Arban', 15000.0, 0),
 (47, '00255', 'Female', 28.0, 'Semi Arban', 84000.0, 0),
 (48, '00256', 'Male', 23.0, 'Semi Arban', 20000.0, 0),
 (49, '00257', 'Male', 25.0, 'Arban', 79000.0, 1),
 (50, '00258', 'Female', 27.0, 'Arban', 54000.0, 0),
 (51, '00259', 'Male', 30.0, 'Arban', 135000.0, 1),
 (52, '00260', 'Female', 31.0, 'Arban', 89000.0, 1),
 (53, '00261', 'Female', 24.0, 'Arban', 32000.0, 1),
 (54, '00262', 'Female', 18.0, 'Arban', 44000.0, 1),
 (55, '00263', 'Female', None, 'Arban', 83000.0, 0),
 (56, '00264', 'Female', 35.0, 'Arban', 23000.0, 1),
 (57, '00265', 'Female', 27.0, 'Arban', 58000.0, 1),
 (58, '00266', 'Female', 24.0, 'Arban', 55000.0, 1),
 (59, '00267', 'Female', 23.0, 'Arban', 48000.0, 1),
 (60, '00268', 'Male', 28.0, 'Rural', 79000.0, 1),
 (61, '00269', 'Male', 22.0, 'Rural', 18000.0, 0),
 (62, '00270', 'Female', 32.0, 'Rural', 117000.0, 1),
 (63, '00271', 'Male', 27.0, 'Rural', 20000.0, 0),
 (64, '00272', 'Male', 25.0, 'Rural', 87000.0, 1),
 (65, '00273', 'Female', 23.0, 'Rural', 66000.0, 0),
 (66, '00274', 'Male', 32.0, 'Semi Arban', 120000.0, 1),
 (67, '00275', 'Female', None, 'Rural', 83000.0, 0),
 (68, '00276', 'Male', 24.0, 'Rural', 58000.0, 1),
 (69, '00277', 'Male', 24.0, 'Rural', 19000.0, 0),
 (70, '00278', 'Female', 23.0, 'Rural', 82000.0, 1),
 (71, '00279', 'Female', 22.0, 'Rural', 63000.0, 1),
 (72, '00280', 'Female', 31.0, 'Semi Arban', 68000.0, 0),
 (73, '00281', 'Male', 25.0, 'Semi Arban', 80000.0, 0),
 (74, '00282', 'Female', 24.0, 'Semi Arban', 27000.0, 0),
 (75, '00283', 'Female', 20.0, 'Semi Arban', 23000.0, 0),
 (76, '00284', 'Female', 33.0, 'Semi Arban', 113000.0, 1),
 (77, '00285', 'Male', 32.0, 'Semi Arban', 18000.0, 0),
 (78, '

In [21]:
%%sql

-- Verificando valores distintos na coluna

SELECT DISTINCT GENDER FROM valentine_gift_purchase

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[('Male',), ('Female',)]

In [22]:
%%sql

-- Verifica se tem valores nulo

SELECT COUNT(*) 
FROM valentine_gift_purchase
WHERE Age is null;

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(7,)]

In [23]:
%%sql

-- Verifica se tem valores nulo

SELECT "Customer ID", "Gender", Age, "Location", EstimatedIncome, Purchased 
FROM valentine_gift_purchase
WHERE Age is null;

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[('00223', 'Male', None, 'Rural', 86000.0, 1),
 ('00233', 'Male', None, 'Semi Arban', 41000.0, 0),
 ('00248', 'Male', None, 'Rural', 49000.0, 0),
 ('00263', 'Female', None, 'Arban', 83000.0, 0),
 ('00275', 'Female', None, 'Rural', 83000.0, 0),
 ('00289', 'Female', None, 'Arban', 87000.0, 0),
 ('00302', 'Female', None, 'Rural', 116000.0, 0)]

In [24]:
%%sql

-- Verifica qual é a moda para preenchimento

SELECT Age, COUNT(*) AS moda FROM valentine_gift_purchase
WHERE Age IS NOT NULL
GROUP BY Age
ORDER BY moda DESC
LIMIT 1

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(35.0, 32)]

In [25]:
%%sql

-- Atualiza com a moda os valores nulos

UPDATE valentine_gift_purchase
SET Age = 32
WHERE Age is null

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[]

In [26]:
%%sql

-- Verifica se tem valores nulo

SELECT COUNT(*) 
FROM valentine_gift_purchase
WHERE Age is null;

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(0,)]

In [27]:
%%sql

-- Retorna os valores distintos

SELECT DISTINCT Age FROM valentine_gift_purchase

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(19.0,),
 (35.0,),
 (26.0,),
 (27.0,),
 (32.0,),
 (25.0,),
 (18.0,),
 (29.0,),
 (47.0,),
 (45.0,),
 (46.0,),
 (48.0,),
 (49.0,),
 (31.0,),
 (21.0,),
 (28.0,),
 (33.0,),
 (30.0,),
 (23.0,),
 (24.0,),
 (22.0,),
 (20.0,),
 (34.0,),
 (39.0,),
 (38.0,),
 (37.0,),
 (42.0,),
 (40.0,),
 (36.0,),
 (41.0,),
 (58.0,),
 (55.0,),
 (52.0,),
 (59.0,),
 (60.0,),
 (56.0,),
 (53.0,),
 (50.0,),
 (51.0,),
 (57.0,),
 (44.0,),
 (43.0,),
 (54.0,)]

In [28]:
%%sql

-- Retorna os valores distintos

SELECT DISTINCT Location FROM valentine_gift_purchase

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[('Rural',), ('Semi Arban',), ('Arban',), (None,)]

In [29]:
%%sql

-- Verifica se tem valores nulo

SELECT COUNT(*) 
FROM valentine_gift_purchase
WHERE Location is null

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(5,)]

In [30]:
%%sql

-- DELETA valores nulo

DELETE FROM valentine_gift_purchase
WHERE Location IS NULL;


 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[]

In [31]:
%%sql

-- Retorna os valores distintos

SELECT DISTINCT Location FROM valentine_gift_purchase

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[('Rural',), ('Semi Arban',), ('Arban',)]

In [32]:
%%sql

-- Conta o total de EstimatedIncome.

SELECT EstimatedIncome, COUNT(*) AS total
FROM valentine_gift_purchase
GROUP BY EstimatedIncome 
Order by total desc

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(80000.0, 12),
 (72000.0, 11),
 (79000.0, 10),
 (23000.0, 10),
 (71000.0, 9),
 (59000.0, 9),
 (88000.0, 8),
 (75000.0, 8),
 (74000.0, 7),
 (60000.0, 7),
 (55000.0, 7),
 (89000.0, 6),
 (82000.0, 6),
 (65000.0, 6),
 (61000.0, 6),
 (57000.0, 6),
 (52000.0, 6),
 (50000.0, 6),
 (47000.0, 6),
 (43000.0, 6),
 (42000.0, 6),
 (33000.0, 6),
 (32000.0, 6),
 (22000.0, 6),
 (20000.0, 6),
 (90000.0, 5),
 (87000.0, 5),
 (86000.0, 5),
 (83000.0, 5),
 (36000.0, 5),
 (28000.0, 5),
 (26000.0, 5),
 (144000.0, 4),
 (108000.0, 4),
 (104000.0, 4),
 (96000.0, 4),
 (84000.0, 4),
 (73000.0, 4),
 (70000.0, 4),
 (63000.0, 4),
 (58000.0, 4),
 (54000.0, 4),
 (53000.0, 4),
 (51000.0, 4),
 (44000.0, 4),
 (41000.0, 4),
 (30000.0, 4),
 (25000.0, 4),
 (15000.0, 4),
 (134000.0, 3),
 (130000.0, 3),
 (118000.0, 3),
 (117000.0, 3),
 (113000.0, 3),
 (112000.0, 3),
 (107000.0, 3),
 (77000.0, 3),
 (76000.0, 3),
 (68000.0, 3),
 (49000.0, 3),
 (45000.0, 3),
 (38000.0, 3),
 (34000.0, 3),
 (29000.0, 3),
 (27000.0, 3),
 (18000.0, 3),
 (17000.0, 3),
 (150000.0, 2),
 (149000.0, 2),
 (146000.0, 2),
 (143000.0, 2),
 (141000.0, 2),
 (138000.0, 2),
 (137000.0, 2),
 (135000.0, 2),
 (133000.0, 2),
 (131000.0, 2),
 (125000.0, 2),
 (122000.0, 2),
 (85000.0, 2),
 (81000.0, 2),
 (78000.0, 2),
 (66000.0, 2),
 (64000.0, 2),
 (62000.0, 2),
 (46000.0, 2),
 (39000.0, 2),
 (31000.0, 2),
 (21000.0, 2),
 (19000.0, 2),
 (16000.0, 2),
 (148000.0, 1),
 (147000.0, 1),
 (142000.0, 1),
 (129000.0, 1),
 (126000.0, 1),
 (123000.0, 1),
 (120000.0, 1),
 (119000.0, 1),
 (116000.0, 1),
 (115000.0, 1),
 (114000.0, 1),
 (106000.0, 1),
 (105000.0, 1),
 (102000.0, 1),
 (101000.0, 1),
 (100000.0, 1),
 (99000.0, 1),
 (97000.0, 1),
 (95000.0, 1),
 (93000.0, 1),
 (91000.0, 1),
 (69000.0, 1),
 (48000.0, 1),
 (37000.0, 1),
 (35000.0, 1)]

In [33]:
%%sql

-- Retorna os valores distintos

SELECT DISTINCT EstimatedIncome FROM valentine_gift_purchase

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(22000.0,),
 (20000.0,),
 (36000.0,),
 (57000.0,),
 (76000.0,),
 (60000.0,),
 (84000.0,),
 (150000.0,),
 (33000.0,),
 (65000.0,),
 (80000.0,),
 (52000.0,),
 (86000.0,),
 (19000.0,),
 (82000.0,),
 (25000.0,),
 (26000.0,),
 (28000.0,),
 (29000.0,),
 (49000.0,),
 (41000.0,),
 (23000.0,),
 (30000.0,),
 (43000.0,),
 (18000.0,),
 (74000.0,),
 (137000.0,),
 (16000.0,),
 (44000.0,),
 (90000.0,),
 (27000.0,),
 (72000.0,),
 (31000.0,),
 (17000.0,),
 (51000.0,),
 (108000.0,),
 (15000.0,),
 (79000.0,),
 (54000.0,),
 (135000.0,),
 (89000.0,),
 (32000.0,),
 (83000.0,),
 (58000.0,),
 (55000.0,),
 (48000.0,),
 (117000.0,),
 (87000.0,),
 (66000.0,),
 (120000.0,),
 (63000.0,),
 (68000.0,),
 (113000.0,),
 (112000.0,),
 (42000.0,),
 (88000.0,),
 (62000.0,),
 (118000.0,),
 (85000.0,),
 (81000.0,),
 (50000.0,),
 (116000.0,),
 (123000.0,),
 (73000.0,),
 (37000.0,),
 (59000.0,),
 (149000.0,),
 (21000.0,),
 (35000.0,),
 (71000.0,),
 (61000.0,),
 (75000.0,),
 (53000.0,),
 (107000.0,),
 (96000.0,),
 (45000.0,),
 (47000.0,),
 (100000.0,),
 (38000.0,),
 (69000.0,),
 (148000.0,),
 (115000.0,),
 (34000.0,),
 (70000.0,),
 (101000.0,),
 (130000.0,),
 (114000.0,),
 (142000.0,),
 (78000.0,),
 (143000.0,),
 (91000.0,),
 (144000.0,),
 (102000.0,),
 (126000.0,),
 (133000.0,),
 (147000.0,),
 (104000.0,),
 (146000.0,),
 (122000.0,),
 (97000.0,),
 (39000.0,),
 (134000.0,),
 (95000.0,),
 (131000.0,),
 (77000.0,),
 (125000.0,),
 (106000.0,),
 (141000.0,),
 (93000.0,),
 (138000.0,),
 (119000.0,),
 (105000.0,),
 (99000.0,),
 (129000.0,),
 (46000.0,),
 (64000.0,)]

In [34]:
%%sql

-- Retorna os valores distintos

SELECT DISTINCT Purchased FROM valentine_gift_purchase

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(0,), (1,)]

In [35]:
%%sql

-- Conta o total de valores Purchased

SELECT Purchased, COUNT(*) AS total
FROM valentine_gift_purchase
GROUP BY Purchased 
Order by total desc

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(1, 231), (0, 181)]

In [36]:
%%sql

-- Verifica se tem valores nulo

SELECT COUNT(*) 
FROM valentine_gift_purchase
WHERE Purchased is null 

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(0,)]

![Valentine Days](imagem\valentine.png)
# <p align="center"><strong>Perguntas abaixo que deverão ser respondidas</strong></p>

- Quais é o gênero que mais compram?
- Como os números de compras estão divididos entre as faixas etárias de 18 a 25, 26 a 35, 36 a 45 e acima de 45?
- A localização influência na hora da compra?
- Renda estimada influência na hora da compra? 
- Qual é a média renda estimada por localização das pessoas que compram?

In [37]:
%%sql

SELECT Purchased, COUNT(*) FROM valentine_gift_purchase

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(0, 412)]

In [38]:
%%sql

SELECT gender, COUNT(*) AS Total_generos_compram FROM valentine_gift_purchase
WHERE Purchased = 1
GROUP BY gender
ORDER BY Total_generos_compram DESC;


 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[('Male', 151), ('Female', 80)]

In [39]:
%%sql

SELECT 
    CASE 
        WHEN Age >= 18 AND Age <= 25 THEN  'Faixa etária de 18 a 25'
        WHEN Age >= 26 AND Age <= 35 THEN  'Faixa etária de 26 a 35'
        WHEN Age >= 36 AND Age <= 45 THEN  'Faixa etária de 36 a 45'
        WHEN Age > 45 THEN 'Faixa etária acima de 45'
    END AS faixa_etaria,
    COUNT(*) AS total_pessoas_por_faixa_etaria
FROM valentine_gift_purchase
WHERE Purchased = 1
GROUP BY 
    CASE 
        WHEN Age >= 18 AND Age <= 25 THEN  'Faixa etária de 18 a 25'
        WHEN Age >= 26 AND Age <= 35 THEN  'Faixa etária de 26 a 35'
        WHEN Age >= 36 AND Age <= 45 THEN  'Faixa etária de 36 a 45'
        WHEN Age > 45 THEN  'Faixa etária acima de 45'
    END
   

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[('Faixa etária acima de 45', 55),
 ('Faixa etária de 18 a 25', 20),
 ('Faixa etária de 26 a 35', 74),
 ('Faixa etária de 36 a 45', 82)]

In [40]:
%%sql

SELECT Location, Purchased, COUNT(*) AS Total_de_compras_por_localizacao FROM valentine_gift_purchase
WHERE Purchased = 1
GROUP BY Location, Purchased
ORDER BY Total_de_compras_por_localizacao DESC

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[('Arban', 1, 129), ('Semi Arban', 1, 58), ('Rural', 1, 44)]

In [41]:
%%sql

SELECT EstimatedIncome, Purchased, COUNT(*) AS Total_de_compras_por_Renda_estimada FROM valentine_gift_purchase
WHERE Purchased = 1
GROUP BY EstimatedIncome, Purchased
ORDER BY Total_de_compras_por_Renda_estimada DESC

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(80000.0, 1, 8),
 (71000.0, 1, 7),
 (88000.0, 1, 7),
 (59000.0, 1, 6),
 (75000.0, 1, 6),
 (79000.0, 1, 6),
 (23000.0, 1, 5),
 (32000.0, 1, 5),
 (55000.0, 1, 5),
 (57000.0, 1, 4),
 (58000.0, 1, 4),
 (60000.0, 1, 4),
 (72000.0, 1, 4),
 (73000.0, 1, 4),
 (74000.0, 1, 4),
 (86000.0, 1, 4),
 (87000.0, 1, 4),
 (89000.0, 1, 4),
 (90000.0, 1, 4),
 (144000.0, 1, 4),
 (22000.0, 1, 3),
 (33000.0, 1, 3),
 (43000.0, 1, 3),
 (52000.0, 1, 3),
 (53000.0, 1, 3),
 (61000.0, 1, 3),
 (65000.0, 1, 3),
 (77000.0, 1, 3),
 (82000.0, 1, 3),
 (83000.0, 1, 3),
 (96000.0, 1, 3),
 (107000.0, 1, 3),
 (108000.0, 1, 3),
 (112000.0, 1, 3),
 (113000.0, 1, 3),
 (117000.0, 1, 3),
 (118000.0, 1, 3),
 (130000.0, 1, 3),
 (28000.0, 1, 2),
 (42000.0, 1, 2),
 (45000.0, 1, 2),
 (50000.0, 1, 2),
 (63000.0, 1, 2),
 (64000.0, 1, 2),
 (70000.0, 1, 2),
 (81000.0, 1, 2),
 (84000.0, 1, 2),
 (104000.0, 1, 2),
 (122000.0, 1, 2),
 (131000.0, 1, 2),
 (134000.0, 1, 2),
 (135000.0, 1, 2),
 (137000.0, 1, 2),
 (138000.0, 1, 2),
 (146000.0, 1, 2),
 (149000.0, 1, 2),
 (150000.0, 1, 2),
 (19000.0, 1, 1),
 (20000.0, 1, 1),
 (25000.0, 1, 1),
 (26000.0, 1, 1),
 (30000.0, 1, 1),
 (31000.0, 1, 1),
 (38000.0, 1, 1),
 (39000.0, 1, 1),
 (44000.0, 1, 1),
 (48000.0, 1, 1),
 (51000.0, 1, 1),
 (54000.0, 1, 1),
 (62000.0, 1, 1),
 (66000.0, 1, 1),
 (68000.0, 1, 1),
 (69000.0, 1, 1),
 (76000.0, 1, 1),
 (78000.0, 1, 1),
 (85000.0, 1, 1),
 (91000.0, 1, 1),
 (93000.0, 1, 1),
 (97000.0, 1, 1),
 (99000.0, 1, 1),
 (100000.0, 1, 1),
 (101000.0, 1, 1),
 (105000.0, 1, 1),
 (106000.0, 1, 1),
 (115000.0, 1, 1),
 (119000.0, 1, 1),
 (120000.0, 1, 1),
 (123000.0, 1, 1),
 (125000.0, 1, 1),
 (126000.0, 1, 1),
 (129000.0, 1, 1),
 (133000.0, 1, 1),
 (141000.0, 1, 1),
 (142000.0, 1, 1),
 (143000.0, 1, 1),
 (147000.0, 1, 1),
 (148000.0, 1, 1)]

In [42]:

%%sql

SELECT Location, ROUND(AVG(EstimatedIncome),2) AS Media_renda_estimada_por_localização  
FROM valentine_gift_purchase
WHERE Purchased = 1
GROUP BY Location
ORDER BY Media_renda_estimada_por_localização DESC


 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[('Arban', 92062.02), ('Semi Arban', 70103.45), ('Rural', 63477.27)]

In [43]:
%%sql

SELECT * FROM valentine_gift_purchase

 * sqlite:///database/Valentine_Gift_Purchase.db
Done.


[(0, '00211', 'Male', 19.0, 'Rural', 22000.0, 0),
 (1, '00212', 'Male', 35.0, 'Semi Arban', 20000.0, 0),
 (2, '00213', 'Female', 26.0, 'Arban', 36000.0, 0),
 (3, '00214', 'Female', 27.0, 'Rural', 57000.0, 0),
 (4, '00215', 'Male', 19.0, 'Rural', 76000.0, 0),
 (5, '00216', 'Male', 27.0, 'Rural', 60000.0, 1),
 (6, '00217', 'Female', 27.0, 'Rural', 84000.0, 0),
 (7, '00218', 'Female', 32.0, 'Arban', 150000.0, 1),
 (8, '00219', 'Male', 25.0, 'Rural', 33000.0, 1),
 (9, '00220', 'Female', 35.0, 'Rural', 65000.0, 1),
 (10, '00221', 'Female', 26.0, 'Rural', 80000.0, 0),
 (11, '00222', 'Female', 26.0, 'Rural', 52000.0, 0),
 (12, '00223', 'Male', 32.0, 'Rural', 86000.0, 1),
 (13, '00224', 'Male', 32.0, 'Semi Arban', 19000.0, 1),
 (14, '00225', 'Male', 18.0, 'Semi Arban', 82000.0, 1),
 (15, '00226', 'Male', 29.0, 'Semi Arban', 80000.0, 1),
 (17, '00227', 'Male', 47.0, 'Arban', 25000.0, 1),
 (18, '00228', 'Male', 45.0, 'Arban', 26000.0, 1),
 (20, '00229', 'Male', 46.0, 'Arban', 28000.0, 1),
 (22, '00230', 'Female', 48.0, 'Arban', 29000.0, 0),
 (23, '00231', 'Male', 45.0, 'Semi Arban', 22000.0, 1),
 (24, '00232', 'Female', 47.0, 'Semi Arban', 49000.0, 0),
 (25, '00233', 'Male', 32.0, 'Semi Arban', 41000.0, 0),
 (26, '00234', 'Female', 45.0, 'Semi Arban', 22000.0, 0),
 (27, '00235', 'Male', 46.0, 'Arban', 23000.0, 1),
 (28, '00236', 'Male', 47.0, 'Arban', 20000.0, 1),
 (29, '00237', 'Male', 49.0, 'Semi Arban', 28000.0, 1),
 (30, '00238', 'Female', 47.0, 'Rural', 30000.0, 1),
 (31, '00239', 'Male', 29.0, 'Rural', 43000.0, 1),
 (32, '00240', 'Male', 31.0, 'Rural', 18000.0, 0),
 (33, '00241', 'Male', 31.0, 'Rural', 74000.0, 1),
 (34, '00242', 'Female', 27.0, 'Rural', 137000.0, 1),
 (35, '00243', 'Female', 21.0, 'Rural', 16000.0, 0),
 (36, '00244', 'Female', 28.0, 'Rural', 44000.0, 0),
 (37, '00245', 'Male', 27.0, 'Rural', 90000.0, 1),
 (38, '00246', 'Male', 35.0, 'Rural', 27000.0, 0),
 (39, '00247', 'Female', 33.0, 'Rural', 28000.0, 0),
 (40, '00248', 'Male', 32.0, 'Rural', 49000.0, 0),
 (41, '00249', 'Female', 26.0, 'Rural', 72000.0, 0),
 (42, '00250', 'Female', 27.0, 'Rural', 31000.0, 0),
 (43, '00251', 'Female', 27.0, 'Rural', 17000.0, 0),
 (44, '00252', 'Female', 33.0, 'Rural', 51000.0, 0),
 (45, '00253', 'Male', 35.0, 'Rural', 108000.0, 1),
 (46, '00254', 'Male', 30.0, 'Semi Arban', 15000.0, 0),
 (47, '00255', 'Female', 28.0, 'Semi Arban', 84000.0, 0),
 (48, '00256', 'Male', 23.0, 'Semi Arban', 20000.0, 0),
 (49, '00257', 'Male', 25.0, 'Arban', 79000.0, 1),
 (50, '00258', 'Female', 27.0, 'Arban', 54000.0, 0),
 (51, '00259', 'Male', 30.0, 'Arban', 135000.0, 1),
 (52, '00260', 'Female', 31.0, 'Arban', 89000.0, 1),
 (53, '00261', 'Female', 24.0, 'Arban', 32000.0, 1),
 (54, '00262', 'Female', 18.0, 'Arban', 44000.0, 1),
 (55, '00263', 'Female', 32.0, 'Arban', 83000.0, 0),
 (56, '00264', 'Female', 35.0, 'Arban', 23000.0, 1),
 (57, '00265', 'Female', 27.0, 'Arban', 58000.0, 1),
 (58, '00266', 'Female', 24.0, 'Arban', 55000.0, 1),
 (59, '00267', 'Female', 23.0, 'Arban', 48000.0, 1),
 (60, '00268', 'Male', 28.0, 'Rural', 79000.0, 1),
 (61, '00269', 'Male', 22.0, 'Rural', 18000.0, 0),
 (62, '00270', 'Female', 32.0, 'Rural', 117000.0, 1),
 (63, '00271', 'Male', 27.0, 'Rural', 20000.0, 0),
 (64, '00272', 'Male', 25.0, 'Rural', 87000.0, 1),
 (65, '00273', 'Female', 23.0, 'Rural', 66000.0, 0),
 (66, '00274', 'Male', 32.0, 'Semi Arban', 120000.0, 1),
 (67, '00275', 'Female', 32.0, 'Rural', 83000.0, 0),
 (68, '00276', 'Male', 24.0, 'Rural', 58000.0, 1),
 (69, '00277', 'Male', 24.0, 'Rural', 19000.0, 0),
 (70, '00278', 'Female', 23.0, 'Rural', 82000.0, 1),
 (71, '00279', 'Female', 22.0, 'Rural', 63000.0, 1),
 (72, '00280', 'Female', 31.0, 'Semi Arban', 68000.0, 0),
 (73, '00281', 'Male', 25.0, 'Semi Arban', 80000.0, 0),
 (74, '00282', 'Female', 24.0, 'Semi Arban', 27000.0, 0),
 (75, '00283', 'Female', 20.0, 'Semi Arban', 23000.0, 0),
 (76, '00284', 'Female', 33.0, 'Semi Arban', 113000.0, 1),
 (77, '00285', 'Male', 32.0, 'Semi Arban', 18000.0, 0),
 (78, '

In [44]:
# Abre a conexão com o banco usando o cnn que é o banco mais execute
# que é usada para enviar uma consulta sql a partir da execução
#e é jogado em query o resultado.
query = cnn.execute("SELECT * FROM valentine_gift_purchase")
query

In [45]:
# No list vai fazer uma iteração sobre cada coluna na descrição do resultado.

# coluna[0]: Para cada coluna, [0] é usado para obter o primeiro elemento. 
# No caso da descrição da coluna, esse primeiro elemento é o nome da coluna
cols = [coluna[0] for coluna in query.description]
cols

['index',
 'Customer ID',
 'Gender',
 'Age',
 'Location',
 'EstimatedIncome',
 'Purchased']

In [46]:
# Gera o dataframe

# Portanto, após a execução desse trecho de código, a variável resultado conterá um DataFrame do pandas com os resultados da
# consulta SQL, e as colunas serão rotuladas de acordo com os nomes extraídos da descrição da consulta.
resultado = pd.DataFrame.from_records(data = query.fetchall(), columns = cols)

In [47]:
# Shape
resultado.shape

(412, 7)

In [48]:
# Visualiza
resultado.head()

index Customer ID  Gender   Age    Location  EstimatedIncome  Purchased
0      0       00211    Male  19.0       Rural          22000.0          0
1      1       00212    Male  35.0  Semi Arban          20000.0          0
2      2       00213  Female  26.0       Arban          36000.0          0
3      3       00214  Female  27.0       Rural          57000.0          0
4      4       00215    Male  19.0       Rural          76000.0          0

In [49]:
# Salva em CSV
resultado.to_csv('database/Valentine_Gift_Purchase_UPDATE.csv', index = False)

![Valentine Days](imagem\valentine.png)

# Conclusão

### Respostas às perguntas de negócio.

- Durante o Valentine's Day, observamos uma tendência marcante em relação ao gênero dos compradores, com o público masculino liderando as compras de presentes. 
- A faixa etária das pessoas que mais compram presentes no Valentine's Day é de 36 a 45 anos. Um fato importante é que a partir dos 26 anos, o volume de compra de presentes aumenta significativamente.
- A localização geográfica também desempenha um papel crucial nas tendências de compra durante o Valentine's Day. As áreas urbanas demonstram um volume considerável de compras em comparação com áreas semi urbanas e rurais. 
- Embora a renda dos consumidores varie, os dados sugerem que não há um padrão claro em relação à renda e às decisões de compra durante o Valentine's Day. 
- Analisando a média de renda por localização de pessoas que compram presente no Valentine's Day, observamos que os consumidores em áreas urbanas têm uma renda média mais alta em comparação com áreas semi urbanas e rurais. locais urbano é de 92062.02, Semi urbano de 70103.45 e Rural de 63477.27.


### Ações
O número de compras realizadas pelo gênero feminino é muito baixo, pois, segundo pesquisas, as mulheres tendem a gastar mais em compras online e em lojas físicas do que os homens. O estudo observou que 76% das mulheres pesquisadas relataram fazer compras online pelo menos uma vez por mês, em comparação com 66% dos homens. Além disso, as mulheres também gastaram mais por transação, em média. O relatório destacou que as mulheres são responsáveis por cerca de 70% das compras globais. Esta é uma descoberta interessante que pode direcionar estratégias de marketing e produtos para atender às preferências desse grupo específico, já que o potencial de compra das mulheres é maior do que o dos homens, e na análise feita, isso é o oposto, ficando muito aquém em relação às compras dos homens.

As gerações Millennials (nascidos entre 1981 e 1996) e a Geração Z (nascidos a partir de meados dos anos 1990 até meados dos anos 2010) tendem a ser os principais compradores online. O dataset oferece um número de compradores nessa faixa etária no Valentine Day's muito semelhante ao potencial fornecido no ambiente online. Assim, criar websites e melhorar a publicidade para pessoas dessas gerações pode aumentar drasticamente as vendas.

Analisando a média de renda por localização, observamos que os consumidores em áreas urbanas têm uma renda média mais alta em comparação com áreas semiurbanas e rurais. Essa disparidade na renda pode influenciar os padrões de gastos e as preferências de compra dos consumidores em diferentes regiões.

- 1: Utilize os dados de renda por localização para segmentar o mercado de forma mais precisa. Desenvolva estratégias de marketing específicas para atender às necessidades e preferências dos consumidores em diferentes regiões, levando em consideração seus níveis de renda.

- 2: A precificação deve ser feita de acordo com as características socioeconômicas das regiões. Isso pode envolver a oferta de preços diferenciados ou pacotes de produtos adaptados às diferentes faixas de renda.

- 3: Ajuste as preferências de compra dos consumidores em cada localização e adapte os produtos e serviços oferecidos para atender às suas necessidades específicas.
